### ISSUE TITLE
Provide an LSP extension method to simply get the type of the thing at a given position.

### ISSUE THREAD

**Issue by no-more-secrets** [clangd/clangd#1658]
Very often I want to say to clangd "give me the type of whatever is under the cursor, if it has a type".  Be it a variable, auto, function, etc.  But all that clangd (and LSP in general) seem to be able to do is to provide a blob of markdown text in response to the "Hover" request.  But I think that is really not ideal since it is visually more difficult to parse and more difficult to deal with programatically; e.g. I don't want to have to parse markdown text to extract the type of a variable.  So I would propose something like:

"textDocument/clangd.getType"

It returns only the type of the thing under the cursor; no comments/docs/etc.

**Comment by HighCommander4** [comment]
A somewhat related feature that already exists is [textDocument/typeDefinition](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_typeDefinition), made available in VSCode under the context menu item "Go to Type Definition".

This doesn't meet your use case exactly in that it performs navigation (which you then have to undo with "Go Back" if you didn't actually want to navigate to the type definition), but the target of the navigation will be the definition of the type you're looking for.

**Comment by no-more-secrets** [comment]
@HighCommander4 Yes I am aware of the typeDefinition... it is sometimes useful, but not really what I'm looking for here as you mentioned.

If we had such a method then I think it would benefit e.g. the YouCompleteMe vim plugin, which currently has to retrieve the Hover markdown text from clangd and then parse it (in kind of a hacky way) to obtain a concise one-line summary of the type, which it provides via its convenient `YcmGetType` command.
